packages/frontend-api/src/Model/User/FrontendApiCustomerUserProvider.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrontendApiBundle\Model\User;
+
+use Lcobucci\JWT\Token;
+use Symfony\Component\Security\Core\User\UserInterface;
+use Symfony\Component\Security\Core\User\UserProviderInterface;
+
+class FrontendApiCustomerUserProvider implements UserProviderInterface
+{
+    /**
+     * @var \Shopsys\FrontendApiBundle\Model\User\FrontendApiUserFactoryInterface
+     */
+    protected $frontendApiUserFactory;
+
+    /**
+     * @param \Shopsys\FrontendApiBundle\Model\User\FrontendApiUserFactoryInterface $frontendApiUserFactory
+     */
+    public function __construct(FrontendApiUserFactoryInterface $frontendApiUserFactory)
+    {
+        $this->frontendApiUserFactory = $frontendApiUserFactory;
+    }
+
+    /**
+     * @param \Lcobucci\JWT\Token $token
+     * @return \Shopsys\FrontendApiBundle\Model\User\FrontendApiUser
+     */
+    public function loadUserFromToken(Token $token): FrontendApiUser
+    {
+        return $this->frontendApiUserFactory->createFromToken($token);
+    }
+
+    /**
+     * @param mixed $username
+     */
+    public function loadUserByUsername($username)
```

Comment:
you should throw exception or something, if you do not support that applies even for next method

Reply:
659f59f9399f1da33c3ec4d7f55d12afc9274efb